Core/Debugger/WebSocket.cpp

```diff
@@ -186,10 +191,19 @@ void HandleDebuggerRequest(const http::Request &request) {
 	while (ws->Process(highActivity ? 1.0f / 1000.0f : 1.0f / 60.0f)) {
 		std::lock_guard<std::mutex> guard(lifecycleLock);
 		// These send events that aren't just responses to requests.
-		logger.Broadcast(ws);
-		game.Broadcast(ws);
-		stepping.Broadcast(ws);
-		input.Broadcast(ws);
+
+		if (allowed_config.at("logger")) {
```

Comment:
Is this useful enough to justify all the extra lines one would need to add and chances of errors/mistakes/confusion when adding another message?

Most useful clients will simply only pay attention to the events they want.  If there were any high bandwidth or expensive messages here, it might make sense, but generally this change will only help a sloppily written client that fails to be able to filter messages itself, while increasing the maintenance work required for future changes of the API.

-[Unknown]

Reply:
1) These extra lines are insignificant next to the quantity of code one would have to add in order to implement a new broadcaster. The exact number is... three lines of code, but if you prefer "if one-liners", it's a single new line (one in the WebsocketClientInfo constructor). As for the errors/mistakes/confusion, I think the implementation is very straightforward.
2) The number of fully functional and useful clients is... eh... one? (Possibly more, if you include some private tools used in gaming/modding communities, but I'm not aware of any other than [this](https://github.com/Nemoumbra/Patapon-Utilities/tree/main/Owocek%20P3_helper)). If the devs think that such design benefits them, for instance, makes them feel safer about their connection or limits, I won't judge. The maintenance work is not substantially increased either. It doesn't even affect the subscribers system which makes up the most of our current API.
3) It's easy to find all the places that require a change. If someone can do it in even fewer number of lines than 1, I'd like to hear how.

Reply:
It's not about how many total lines, it's about how obvious those lines are and how many places.

One example would simply be fixing the bool values.  If the default was false and it was zero-initialized, that'd be one less place to remember to change to have a correct implementation.  The goal with things like this should be easy for "Nemoumbra from 6 months in the future" or "unknownbrackets from 6 months in the future" to write correct code.  We silly humans tend to miss things when they are spread out in multiple places.

If it absolutely cannot be architected simpler, at least it could have a comment.  This does look like it could be simplified to me in multiple ways, though.

-[Unknown]